tensorboard/plugins/debugger/interactive_debugger_plugin_test.py

```diff
@@ -44,11 +44,13 @@
 from tensorboard.plugins.debugger import interactive_debugger_plugin
 from tensorboard.util import test_util
 
+tf = tensorflow.compat.v1
+tf.disable_v2_behavior()
```

Comment:
Interesting: simply adding `tf.compat.v1.disable_v2_behavior()` without
the rest of the patch doesn’t suffice to fix the timeouts on my end, so
some of the API differences must(?) be causing these unbounded
timeouts—do we know which ones?

In the short term, this patch is great because it lets us continue to
run the tests, but ideally we’d like to get rid of `tf.compat.v1`
wherever possible, too (#1718).

Reply:
I added a comment here. The Debugger Plugin is currently tied to v1 behavior (`tf.Session`s), which is the reason why it is necessary to pin V1 behavior here. When tfdbg V2 comes along, this will of course no longer be necessary.

Reply:
Right, that makes sense. But if session and node behavior were the only
salient changes then we would expect adding `disable_v2_behavior()` to
fix the tests, right? And yet if we make only that change then the
`testMultipleSessionRunsTensorValueFullHistory` test case hangs at
various points—sometimes in the `listen` call in `setUp`, sometimes in
the first call to `_serverGet` in the test body. I was wondering why
that is, and why the remaining changes, which just change `tf.foo` to
`tf.compat.v1.foo`, _do_ fix the tests.

(This isn’t necessarily a blocking question, but I would be interested
to learn how this PR actually works.)

Reply:
I confirmed what you said. It seems that making only the change of
1. Using `import tensorflow.compat.v1 as tf` 
2. Using `tf.disable_v2_behavior()`

leads to hanging tests. This looks like some unexpected behavior in how `tf.compat.v1.compat.v1....` behaves. 

As you said, this doesn't have to be blocking. I can make a TODO item for myself to investigate it and potentially file  bug for tensorflow while merging this PR. LMK WYT.

Reply:
Using `tf.compat.v1.compat.v1` will simply not work, in either TF 1.x or
TF 2.x. But making _only_ the `disable_v2_behavior()` change seems like
it _should_ work, because the rest of the code already uses
`tf.compat.v1.Session()` explicitly.

This is good to merge; feel free to investigate iff you feel like it.

Reply:
`disable_v2_behavior()` exists only in `tf` from TF1 or `tf.compat.v1` from TF2. So to use it, we'd have to do `import tf.compat.v1 as tf` regardless (TF1 tests are passing, so they are out of concern). But if you use that and don't change anything below, you'll just end up using `tf.compat.v1.compat.v1.*`, which I think is what causes the hanging. 

I'll merge the PR. Thanks for the review.

Reply:
Let me be concrete: the patch that I’m talking about is simply

```diff
diff --git a/tensorboard/plugins/debugger/interactive_debugger_plugin_test.py b/tensorboard/plugins/debugger/interactive_debugger_plugin_test.py
index 1b127456..653af030 100644
--- a/tensorboard/plugins/debugger/interactive_debugger_plugin_test.py
+++ b/tensorboard/plugins/debugger/interactive_debugger_plugin_test.py
@@ -45,6 +45,9 @@ from tensorboard.plugins.debugger import interactive_debugger_plugin
 from tensorboard.util import test_util
 
 
+tf.compat.v1.disable_v2_behavior()
+
+
 _SERVER_URL_PREFIX = '/data/plugin/debugger/'
 
 
```

against `master` prior to this PR (i.e., 2b96c2a1).

This properly disables V2 behavior, does not use `compat.v1.compat.v1`,
and uses `tf.compat.v1.Summary`, but still hangs.

Reply:
And `tf.compat.v1.compat.v1` is a hard error, not a loop.

Reply:
Acknowledged. Thanks for the concrete info.